### PR TITLE
Threads

### PR DESCRIPTION
**Pull request by panosfol**
This PR changes the way that we handle the creation of the log file and adds thread functionality.

### CODE REVIEW

**Review by charmitro — CHANGES_REQUESTED** [review]
Some review for starters.

File: src/db.c [hunk 1]
```diff
@@ -16,46 +18,70 @@ int num_init_functions = 0;
 struct db_operations **available_dbs;
 size_t db_ops_counter = 0;
 
+void *db_thread(void *arg)
```
Comment (charmitro) [inline src/db.c:21]:
```suggestion
void *db_operation_thread(void *arg)
```

File: include/db/db.h [hunk 2]
```diff
@@ -1,8 +1,10 @@
 #ifndef DB_H
 #define DB_H
 
-#include "config.h"
 #include <stdbool.h>
```
Comment (charmitro) [inline include/db/db.h:4]:
```suggestion
```

File: src/log.c [hunk 3]
```diff
@@ -13,21 +14,18 @@
 
 bool verbose = false;
 
-FILE *log_file;
 extern config_t *ini_config;
+extern char *log_filepath;
 
-void construct_log_filename(char **log_filename, char *log_filepath)
+void construct_log_filename(struct db_t *db_t, const char *log_name)
```
Comment (charmitro) [inline src/log.c:20]: Why this change? Does `construct_log_filename` for something other than the filename?

**Review by charmitro — CHANGES_REQUESTED** [review]
This can be removed now, right?

https://github.com/terra-sync/cnc/blob/170b9ad82343084fc4bc1b1a312b363bc66e4c39/include/log.h#L16

File: src/log.c (hunk 3 above)
Comment (panosfol) [reply src/log.c:20]: > Why this change? Does `construct_log_filename` for something other than the filename?

You are right, this was a redundant change, ill fix it.

**Review by charmitro — APPROVED** [review]

**Review by charmitro — COMMENTED** [review]
> This can be removed now, right?
> 
> https://github.com/terra-sync/cnc/blob/170b9ad82343084fc4bc1b1a312b363bc66e4c39/include/log.h#L16

What about this? @panosfol

**Comment by charmitro** [comment]
> > This can be removed now, right?
> > https://github.com/terra-sync/cnc/blob/170b9ad82343084fc4bc1b1a312b363bc66e4c39/include/log.h#L16
> 
> What about this? @panosfol

Correction, here's the leftover extern:

https://github.com/terra-sync/cnc/blob/9342df84aa6f78a950f662014bf6c2a60cade3ac/src/cnc.c#L15

**Review by charmitro — APPROVED** [review]
